File: Add Two Numbers/AddTwoNumbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        result = ListNode(0)
        curr = result
        carry = 0
                
        while l1 or l2 or carry:            
            n1  = (l1.val if l1 else 0)
            n2  = (l2.val if l2 else 0)
            carry, out = divmod(n1 + n2 + carry, 10)    
                      
            curr.next = ListNode(out)
            curr = curr.next                      
            
            l1 = (l1.next if l1 else None)
            l2 = (l2.next if l2 else None)
               
        return result.next

File: Add Two Numbers/AddTwoNumbers.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function (l1, l2) {
    let carry = 0;

    const result = new ListNode();
    let curr = result;

    while (l1.val !== undefined || l2.val !== undefined) {
        if (curr.val !== undefined) {
            curr = curr.next = new ListNode();
        }
        const v1 = l1.val || 0;
        const v2 = l2.val || 0;

        const sum = v1 + v2 + carry;
        carry = 0;

        if (sum >= 10) {
            curr.val = sum - 10;
            carry = 1;
        } else {
            curr.val = sum;
        }
        l1 = l1.next || {};
        l2 = l2.next || {};
    }

    if (carry) {
        curr.next = new ListNode(1)
    }

    return result;
};